src-tauri/src/image_analyzer.rs

```diff
@@ -243,10 +259,19 @@ impl ImageAnalyzer {
         }
     }
 
-    pub fn detect_hp(&self, last_hp: Hp) -> Option<Hp> {
-        const MAX_SEARCH_X: u32 = 310;
-        const MAX_SEARCH_Y: u32 = 120;
-        let refs: [[u8; 3]; 4] = [[174, 18, 55], [188, 24, 62], [204, 30, 70], [220, 36, 78]];
+    pub fn detect_stats_bar(&self, last_hp: Stat,bar_name:&str) -> Option<Stat> {
+       let bar_option: Option<Bar> = match bar_name  {
+            "HP" => Some(HP_BAR),
+            "MP" => Some(MP_BAR),
+            "FP" => Some(FP_BAR),
+            "EXP" => Some(EXP_BAR),
+            &_ => None,
+        };
```

Comment:
Instead of having the `bar_name` as a string you can use an enum:
```rust
#[derive(Debug)]
pub enum StatusBar {
    Hp,
    Mp,
    Fp,
    Exp,
}
```

Then you can match on that:
```rust
let status_bar_config = match status_bar {
    StatusBar::Hp => HP_BAR,
    StatusBar::Mp => MP_BAR,
    StatusBar::Fp => FP_BAR,
    StatusBar::Exp => XP_BAR,
};
```

That way it's unambiguous which status bar is used and you don't need an `Option`.

Reply:
You could even go one step further and delegate the `Stats` to this enum:
```rust
impl StatusBar {
	fn to_config(&self) -> Stats {
		match self {
			Self::Hp => HP_BAR,
			Self::Fp => FP_BAR,
			Self::Mp => MP_BAR,
			Self::Exp => XP_BAR,
		}
	}
}
```

That way you don't even need to match later and you can just call `let config = status_bar.to_config();`